### PR TITLE
BaSP-TG-54: redux and modal implementation on task list

### PR DESCRIPTION
**Pull request by lucasrendo**
- New feature: confirmation modal before deleting a task
- New feature: informational modal after deleting a Task with status message
- Updated: most React states were replaced by redux states
- Update: All fetchs are now done through redux middleware, outside the list index file

### CODE REVIEW

**Review by GonzaloBracamonte — DISMISSED** [review]
Se ve bien y funciona correctamente, un trabajo genial!

**Review by CorinaArreguez — APPROVED** [review]
Funciona perfecto!

File: src/redux/Task/actions.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import {
+  GET_TASKS_PENDING,
+  GET_TASKS_FULFILLED,
```
Comment (Julian25) [inline src/redux/Task/actions.js:3]: cambiar fullfiled por success (en todos los casos)

**Review by nicomicheletti — APPROVED** [review]

**Review by nicomicheletti — APPROVED** [review]
Correcciones implementadas

**Review by negrojcb — APPROVED** [review]

File: src/redux/Task/constants.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export const GET_TASKS_PENDING = 'GET_TASKS_PENDING';
+export const GET_TASKS_SUCCESS = 'GET_TASKS_SUCCESS';
+export const GET_TASKS_FAILED = 'GET_TASKS_FAILED';
```
Comment (BaltazarCortina1) [inline src/redux/Task/constants.js:3]: El caso de error es `ERROR`, no `FAILED`

File: src/Components/Tasks/List/index.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  const formatTaskData = (responseData) => {
+    const data = responseData.map((task) => {
+      return {
+        id: task._id,
+        employee: task.employeeId ? task.employeeId.firstName + ' ' + task.employeeId.lastName : '',
+        project: task.projectId ? task.projectId.projectName : '',
+        title: task.title,
+        description: task.description,
+        date: task.date.slice(0, 10),
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/List/index.jsx:40]: Deberias validar que date exista

File: src/Components/Tasks/List/index.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        done: task.done.toString()
+      };
+    });
+    return data;
+  };
+
+  const confirmationHandler = () => {
+    setConfirmation(false);
+    dispatch(deleteTask(id));
+    !error && dispatch(updateList([...list.filter((task) => task._id !== id)]));
```
Comment (BaltazarCortina1) [inline src/Components/Tasks/List/index.jsx:50]: Esto lo deberias hacer en el caso de success del delete

**Pull request merged**